README.md

```diff
@@ -549,29 +576,66 @@ A revocation message MUST conform to the following format:
 
-Any other witnesses in the selected UCAN not issued by the same DID as the revocation issuer MUST be treated as valid.
+Any other proofs in the selected UCAN not issued by the same DID as the revocation issuer MUST be treated as valid.
 
 Revocations MAY be deleted once the UCAN that they reference expires or otherwise becomes invalid via its proactive mechanisms.
 
 ## 5.8 Backwards Compatibility
 
-A UCAN validator MAY implement backward compatibility with previous versions of UCAN. Delegated UCANs MUST be of an equal or higher version than their proofs. For example, a v0.8.1 UCAN that includes witnesses that are separately v0.8.1, v0.8.0, v0.7.0, and v0.5.0 MAY be considered valid. A v0.5.0 UCAN that has a UCAN v0.7.0 witness MUST NOT be considered valid.
+A UCAN validator MAY implement backward compatibility with previous versions of UCAN. Delegated UCANs MUST be of an equal or higher version than their proofs. For example, a v0.9.0 UCAN that includes proofs that are separately v0.9.0, v0.8.1, v0.7.0, and v0.5.0 MAY be considered valid. A v0.5.0 UCAN that has a UCAN v0.9.0 proof MUST NOT be considered valid.
+
+# 6. Collections
+
+UCANs are indexed by their hash — often called their ["content address"](https://en.wikipedia.org/wiki/Content-addressable_storage). UCANs MUST be addressable as [CIDv1](https://specs.ipld.io/block-layer/CID.html#cids-version-1).
+
+
+Content addressing the proofs has multiple advantages over inlining tokens, including:
+* Avoids re-encoding deeply nested proofs as Base64 many times (and the associated size increase)
+* Canonical signature
+* Enables only transmitting the relevant proofs 
+
+
+
+Same:
+> It is unresolved whether an implementation must verify that a CID present in the roots array of the Header also appears as a block in the archive. While it is expected that this would be the case, it is unresolved whether encoders and decoders must validate the existence of root blocks in the archive.
+
+<!-- Canonical Signature on CIDs .... move up nearer top -->
+
+# 7. Token Resolution
+
+
+## NUM.NUM Canonical Representation
+
+## NUM.NUM Collections
+
+Multiple UCANs in a single request MAY be collected into one a hash table. It is RECOMMENDED that these be indexed by 
+
+<!-- FIXME -->
+
+
+Due to the potential for unresolvable CIDs, this SHOULD NOT be the preferred method of transmission. "Inline proofs" SHOULD be used whenever possible, and complete UCANs SHOULD be expanded. When a CID is used, it is RECOMMENDED that it be substituted as close to the top UCAN as possible (i.e. the invocation), and as few proofs are referenced by CID, to keep the number of required CID resolutions to a minimum. As UCANs are signed, all further delegations would require CID resolution and so SHOULD NOT be used when the intention is delegation rather than invocation. 
```

Comment:
ah wait think this answers my above question

im very in support of removing the DHT it's a huge performance overhead.

buuttt, i'm a bit confused how the transfer actually happens (am i missing it somewhere or is it not specified?)

UCANs are nice cause you can stick them in the authorization header, but you (probably) can't stick a whole CAR file in an auth header and putting it in the req body seems out of place. and even more out of place in query params in a GET request.

we're using UCANs both _on_ the data structure, and sent in delegation requests to a data server. in the former, it makes sense to just stick all the proofs in the CAR file for the data structure. But for delegation requests - where we use UCANs as a Bearer token - I'm not sure where to put the supporting proofs

sorry if i'm missing this somewhere, I need to play a bit of catch up work on the spec 😅 (coming soon once we finish this prototype!)

Reply:
> buuttt, i'm a bit confused how the transfer actually happens (am i missing it somewhere or is it not specified?)

It is not yet specified! Current thinking is that the transport would be broken out into "UCAN over HTTP", "UCAN over IPFS", "UCAN over mycustomtransport" specs, but this spec would constrain the kind of info that would come across in that request.

Reply:
Maybe extra context: I know that you're already looking at the solution space, and would like to make sure that we support (and maybe even integrate or generalize) what you have

Reply:
okay cool cool 👍 

The motivation makes a lot of sense. especially reading your comment somewhere about re-encoding base64 data & how that adds 30% blowup on each time. and if you have a long UCAN chain of device -> device attenuation, this really simplifies the negotiation process with hosts you're familiar with because they've seen that chain before & only need to check the new ones.

---

so to expand on what i mentioned above:
we have a couple of different "client types", 
- full client has the entire user's repo downloaded (all data, cids, history, etc)
- light client has some subsection of the user's repo. probably non-historical & just the branches that they are editing.
- delegator client does not download _any_ blocks & delegates operations to a server to act on its behalf - "please add a post for me!"

UCANs for a given operation are directly on the root of the repo.

Full clients & light clients do the operation, stick the necessary ucan on the root, sign it, & send it up. This actually works very well for having proofs as CIDs: just put the UCAN & all supporting proofs in the blockstore/ CAR file on transfer

With delegator clients, we have the user generate a very tightly bound token for the data server (ie "you can add this exact post id to this exact collection for the next 20s"). And we're just sending that along as a Bearer token. We expect these UCANs to be not terribly nested, a couple layers. So inlining proofs works well here because it eliminates any dependencies. No extra CAR file, no extra data in the body, no DHT lookup, no negotiation of HTTP. It would be a shame to lose out on that simplicity

Reply:
that makes sense to me^^ headers seem to be the natural place for them. I hadn't thought of splitting them between separate headers 👍

Reply:
I also really like the multiple-header idea 👍 It seems very "browser native".

My gut says that we'll likely need to specify both header and payload. Servers have different limits to the amount of header that they'll accept, in the 4KB-48KB range. IIRC most are around 8KB, which is _plenty_ for most use cases, but having the ability to handle very long chains can't hurt. _Or maybe I'm prematurely optimizing,_ its possible.